knimin/handlers/ag_search.py

```diff
@@ -58,5 +60,7 @@ def post(self):
             display_results.append(login_display)
 
         # now render the page
-        self.render("ag_search.html", results=display_results,
-                    handouts=handouts, currentuser=self.current_user)
+        self.render("ag_search.html",
+                    results=xhtml_escape_recursive(display_results),
+                    handouts=map(xhtml_escape, handouts),
```

Comment:
@sjanssen2 I'm wondering why the text should be escaped here? These values are used to render the template on the server side, so this ends up going to the user in the body of the html, which shouldn't need to explicitly escaped. Just commenting once - but applies to all calls to "render".